app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,21 @@
+class CommentsController < ApplicationController
+  def create
+    @user = User.find(params[:user_id])
+    @post = @user.posts.find(params[:post_id])
+    @comment = @post.comments.build(comment_params)
+    @comment.user = current_user # Assuming you have user authentication in place
+
```

Comment:
[OPTIONAL] All you need to do is this:
 @comment = @current_user.comments.new(params)
   @comment.post_id = params[:post_id]
   Since you have created the current_user method in the application controller.